pkg/agent/deployer/generic/generic.go

```diff
@@ -126,6 +158,154 @@ func (deployer *Deployer) handleDescription(desc *appsapi.Description) error {
 			deployer.discoveryRESTMapper, desc, deployer.recorder)
 	}
 
-	return utils.ApplyDescription(context.TODO(), deployer.clusternetClient, deployer.dynamicClient,
-		deployer.discoveryRESTMapper, desc, deployer.recorder)
+	return deployer.ApplyDescriptionWithInformer(context.TODO(), desc)
+}
+
+func (deployer *Deployer) ApplyDescriptionWithInformer(ctx context.Context, desc *appsapi.Description) error {
+	var allErrs []error
+	wg := sync.WaitGroup{}
+	objectsToBeDeployed := desc.Spec.Raw
+	errCh := make(chan error, len(objectsToBeDeployed))
+	for _, object := range objectsToBeDeployed {
+		resource := &unstructured.Unstructured{}
+		err := resource.UnmarshalJSON(object)
+		if err != nil {
+			allErrs = append(allErrs, err)
+			msg := fmt.Sprintf("failed to unmarshal resource: %v", err)
+			klog.ErrorDepth(5, msg)
+			deployer.recorder.Event(desc, corev1.EventTypeWarning, "FailedMarshalingResource", msg)
+		} else {
+			labels := resource.GetLabels()
+			if labels == nil {
+				labels = map[string]string{}
+			}
+			labels[known.ObjectControlledByNsLabel] = desc.Namespace
```

Comment:
I don't think these two labels are needed. All the resources are reconciled respectively. A `Description` cloud be ignored after all the resources in it are added to the queue. Another queue will keeps syncing new/updated resources from Description changes.